src/test/regress/sql/function_propagation.sql

```diff
@@ -665,6 +665,19 @@ COMMIT;
 -- Function should be marked as distributed on the worker after committing changes
 SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_in_transaction_def_with_seq'::regproc::oid;$$) ORDER BY 1,2;
 
+SET citus.shard_replication_factor = 1;
+-- test creating a colocated function
+CREATE TABLE tbl_to_colocate (a int);
+SELECT create_distributed_table('tbl_to_colocate', 'a');
+CREATE FUNCTION func_to_colocate (a int) returns int as $$select 1;$$ language sql;
+SELECT create_distributed_function('func_to_colocate(int)','$1','tbl_to_colocate');
+-- see the pg_dist_object entry
+SELECT distribution_argument_index, colocationid, force_delegation FROM citus.pg_dist_object WHERE objid = 'func_to_colocate'::regproc;
+-- try create or replace the same func
+CREATE OR REPLACE FUNCTION func_to_colocate (a int) returns int as $$select 1;$$ language sql;
+-- verify the pg_dist_object entry is the same
+SELECT distribution_argument_index, colocationid, force_delegation FROM citus.pg_dist_object WHERE objid = 'func_to_colocate'::regproc;
+
```

Comment:
Can we add tests to show
i) Converting function from delegated to non-delegated
ii) Converting function from force-delegated to none
iii) Creating a distributed function with CREATE FUNCTION, showing that pg_dist_object doesn't have delegation info, delegating it with create_distributed_function and showing pg_dist_object again to show delegation info (You can even create_distributed_function without parameters to show it'll become non-delegated)
iv) Colocating function with reference table after creating a function with CREATE FUNCTION